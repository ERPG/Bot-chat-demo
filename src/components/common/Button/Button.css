@import '../../Chat//InputWrapper/InputWrapper.css';

.landbot-input-send {
  position: absolute;
  right: 20px;
  top: 50%;
  transform: translateY(-50%);
  color: var(--orange);
  background-color: transparent;
  border: 0;
}

.landbot-input-send:hover {
  color: var(--orange);
}

.button[disabled] {
  background-color: transparent;
  border-color: transparent;
}