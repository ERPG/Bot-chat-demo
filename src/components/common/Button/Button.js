import React from 'react';

import './Button.css';

/**
 * Represents a base Button.
 * @constructor
 * @param {ReactNode} children - An element or text inside button.
 * @param {boolean} isDisabled - disables or not the button.
 * @param {MouseEvent<HTMLButtonElement>} onClick - Action that is executed on click event.
 * There are more props in this element, for the purpose of this test we will use this few.
 */

const Button = ({children, isDisabled, onClick}) => {
  return (
    <>
      <button
        data-landbot="send-button"
        className="button landbot-input-send"
        onClick={onClick}
        disabled={isDisabled}
      >
        {children}
      </button>
    </>
  )
}

export default Button;