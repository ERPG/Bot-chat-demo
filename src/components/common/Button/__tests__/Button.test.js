import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';
import Button from '../Button';

describe('Button component', () => {
  const buttonSelector = 'send-button';
  const buttonProps = {
    onClick: jest.fn(),
    isDisabled: false,
    children: 'button text'
  }
  beforeEach(() => {
    jest.clearAllMocks();
  })

  it('Should render component without errors', () => {
    const { container } = render(<Button {...buttonProps}/>);
    expect(container).toMatchSnapshot();
  });

  it('Should call onclick event', () => {
    const { getByDataLb } = customRender(<Button {...buttonProps} />);
    const button = getByDataLb(buttonSelector)

    fireEvent.click(button)

    expect(buttonProps.onClick).toHaveBeenCalled();
  });

    it('Should not call onclick event on disabled', () => {
    const newProps = {...buttonProps, isDisabled: true }
    const { getByDataLb } = customRender(<Button {...newProps } />);
    const button = getByDataLb(buttonSelector)

    fireEvent.click(button)

    expect(newProps.onClick).not.toHaveBeenCalled();
  });

  it('Should have a content', () => {
    const { getByDataLb } = customRender(<Button {...buttonProps} />);
    const element = getByDataLb(buttonSelector);

    expect(element.textContent).toBe('button text');
  });
});