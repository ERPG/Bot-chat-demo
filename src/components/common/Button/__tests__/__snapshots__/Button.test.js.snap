// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render component without errors 1`] = `
<div>
  <button
    class="button landbot-input-send"
    data-landbot="send-button"
  >
    button text
  </button>
</div>
`;
